.github/workflows/main.yml

```diff
@@ -0,0 +1,140 @@
+name: CI
+
+on: [push, pull_request, release]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        component: [builder, cli, client, machine, serializer, server, util, workflow]
+        python-version: [3.7]
+    steps:
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: 'x64'
+
+      - uses: actions/cache@v1
+        if: startsWith(runner.os, 'Linux')
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}
```

Comment:
When does this save the cache? Magically after the steps are done? And should we add the test req file as well?

Reply:
It does auto add the save step after the last job completes successfully. 
Add a wildcard for all `*requirements.txt` files to be included.